ckb-debugger-tests/src/lib.rs

```diff
@@ -95,3 +97,130 @@ impl From<packed::Script> for ChildScript {
+    let mut r = Vec::<usize>::new();
+
+    let mut group_index = group_index;
+    let mut hash_buf = HashSet::<Byte32>::new();
+
+    let mut cur_script_hash: Option<Byte32> = None;
+
+    for i in 0..repr_tx.mock_info.inputs.len() {
+        let input_cell = repr_tx.mock_info.inputs.get(i).unwrap();
+        let hash = {
+            let script = &input_cell.output.lock;
+            let script = Script::new_builder()
+                .args(script.args.as_bytes().pack())
+                .code_hash(script.code_hash.pack())
+                .hash_type((script.hash_type.clone() as u8).into())
+                .build();
+            script.calc_script_hash()
+        };
+
+        if group_index == 0 {
+            if cur_script_hash.is_none() {
+                cur_script_hash = Some(hash.clone());
+                r.push(i);
+            } else {
+                if cur_script_hash.eq(&Some(hash.clone())) {
+                    r.push(i);
+                }
+            }
+        } else if hash_buf.get(&hash).is_none() {
+            hash_buf.insert(hash.clone());
+            group_index -= 1;
+        }
+    }
+
+    r
+}
+
+pub fn generate_sighash_all(
+    tx: &MockTransaction,
+    group_index: usize,
+) -> Result<[u8; 32], anyhow::Error> {
+    let repr_tx: ReprMockTransaction = tx.clone().into();
+
+    let lock_indexs = get_index_by_group_index(group_index, &repr_tx);
+    if lock_indexs.is_empty() {
+        panic!("not get lock index");
+    }
+
+    let zero_extra_witness = {
+        let mut buf = Vec::new();
+        let witness = tx
+            .tx
+            .witnesses()
+            .get(lock_indexs[0])
+            .expect("index:0 witness");
+        buf.resize(witness.len() - 20, 0);
+        buf
+    };
+
+    let witness_args = packed::WitnessArgsBuilder::default()
+        .lock(Some(Bytes::from(zero_extra_witness)).pack())
+        .build();
+
+    let mut blake2b = new_blake2b();
+    let mut message = [0u8; 32];
+
+    let tx_hash = tx.tx.calc_tx_hash();
+    blake2b.update(&tx_hash.raw_data());
+    // println!("--hash: {:02X?}", &tx_hash.raw_data().to_vec());
+    let witness_data = witness_args.as_bytes();
+    blake2b.update(&(witness_data.len() as u64).to_le_bytes());
+    // println!(
+    //     "--hash: {:02X?}",
+    //     &(witness_data.len() as u64).to_le_bytes()
+    // );
+    blake2b.update(&witness_data);
+    // println!("--hash: {:02X?}", &witness_data.to_vec());
+
+    // group
+    if lock_indexs.len() > 1 {
+        for i in 1..lock_indexs.len() {
+            let zero_extra_witness = {
+                let mut buf = Vec::new();
+                let witness = tx
+                    .tx
+                    .witnesses()
+                    .get(lock_indexs[i])
+                    .expect("index:0 witness");
+                buf.resize(witness.len() - 20, 0);
+                buf
+            };
+
+            let witness_args = packed::WitnessArgsBuilder::default()
```

Comment:
This is zeroing not necessary. See:
https://github.com/nervosnetwork/ckb-system-scripts/blob/e08e6016f16072fc2f44cf889ae063fa5b7e10c7/c/secp256k1_blake160_sighash_all.c#L166-L189